internal/repositories/urls/file_repository.go

```diff
@@ -0,0 +1,69 @@
+package urls
+
+import (
+	"log"
+
+	"github.com/izaake/go-shortener-tpl/internal/models"
+	"github.com/izaake/go-shortener-tpl/internal/services/file"
+)
+
+type fileRepository struct {
+	fileName string
+}
+
+func NewFileRepository(fileName string) Repository {
+	return &fileRepository{fileName: fileName}
+}
+
+func (r fileRepository) Save(user *models.User, ignoreConflicts bool) error {
+	filePath := r.fileName
+	if filePath != "" {
+		err := file.WriteToFile(filePath, user)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r fileRepository) FindOriginalURLByShortURL(url string) string {
```

Comment:
Если в функции внутри возникает ошибка, то ее надо пробрасывать наверх. Там ее можно игнорировать или нет, но гасить на низком уровне ошибку, особенно файловой системы - не следует.
```suggestion
func (r fileRepository) FindOriginalURLByShortURL(url string) (string, error) {
```